Assets/Scripts/Internal/DaggerfallBillboard.cs

```diff
@@ -82,34 +79,31 @@ void Start()
                     // Example is the treasury in Daggerfall castle, some action records flow through the quest item marker
                     meshRenderer.enabled = false;
                 }
-            }
-        }
 
-        void OnDisable()
-        {
-            restartAnims = true;
-        }
+                // wait a single frame so all objects initialize
+                yield return null;
 
-        void Update()
-        {
-            // Restart animation coroutine if not running
-            if (restartAnims && summary.AnimatedMaterial)
-            {
-                StartCoroutine(AnimateBillboard());
-                restartAnims = false;
-            }
+                // Rotate to face camera in game
+                // Do not rotate if MeshRenderer disabled. The player can't see it anyway and this could be a hidden editor marker with child objects.
+                // In the case of hidden editor markers with child treasure objects, we don't want a 3D replacement spinning around like a billboard.
+                // Treasure objects are parented to editor marker in this way as the moving action data for treasure is actually on editor marker parent.
+                // Visible child of treasure objects have their own MeshRenderer and DaggerfallBillboard to apply rotations.
+                if (meshRenderer.enabled)
+                {
+                    if (faceY) DaggerfallBillboardSystem.AddPointBillboard(transform);
+                    else DaggerfallBillboardSystem.AddAxialBillboard(transform);
+                }
 
-            // Rotate to face camera in game
-            // Do not rotate if MeshRenderer disabled. The player can't see it anyway and this could be a hidden editor marker with child objects.
-            // In the case of hidden editor markers with child treasure objects, we don't want a 3D replacement spinning around like a billboard.
-            // Treasure objects are parented to editor marker in this way as the moving action data for treasure is actually on editor marker parent.
-            // Visible child of treasure objects have their own MeshRenderer and DaggerfallBillboard to apply rotations.
-            if (mainCamera && Application.isPlaying && meshRenderer.enabled)
-            {
-                float y = (FaceY) ? mainCamera.transform.forward.y : 0;
-                Vector3 viewDirection = -new Vector3(mainCamera.transform.forward.x, y, mainCamera.transform.forward.z);
-                transform.LookAt(transform.position + viewDirection);
+                // Restart animation coroutine if not running
+                if (summary.AnimatedMaterial)
+                    StartCoroutine(AnimateBillboard());
             }
+            else yield return null;
```

Comment:
Why wait the extra frame there?

Reply:
[49ae337](https://github.com/Interkarma/daggerfall-unity/pull/2389/commits/49ae33762506451b4d9f93d88513da3608822a9c) fixed